src/Microsoft.ML.AutoML/TransformInference/TransformInference.cs

```diff
@@ -180,6 +183,29 @@ public override IEnumerable<SuggestedTransform> Apply(IntermediateColumn[] colum
                 }
             }
 
+            internal sealed class LabelCategorical : TransformInferenceExpertBase
+            {
+                public LabelCategorical(MLContext context) : base(context)
+                {
+                }
+
+                public override IEnumerable<SuggestedTransform> Apply(IntermediateColumn[] columns, TaskKind task)
+                {
+                    foreach (var column in columns)
+                    {
+                        if (column.Purpose != ColumnPurpose.LabelFeature)
+                        {
+                            continue;
+                        }
+
+                        if (!column.Type.IsKey())
+                        {
+                            yield return ValueToKeyMappingExtension.CreateSuggestedTransform(Context, column.ColumnName, column.ColumnName);
+                        }
+                    }
+                }
+            }
```

Comment:
Somewhere earlier, we should push the MatrixColumnIndexColumnName in to the Columns.SamplingKey, then handle it nearby here.

Specifically, this causes the dataset to split on the the UserId when creating either the CV or TrainTest splits. Data leakage other occurs as part of the same user's data is in the train set and part is in the test dataset. We should test on distinct users not seen in the training set.

If no splitting occurs in the AutoML code, for instance if a train/validate dataset is explicitly provided, this added part will have no impact (which is good).

Reply:
We should also allow the user to set a different `SamplingKey`, or empty it should they want to. 

For instance, they may want to enforce that their model can be predictive across geographic boundaries, so they may pick their `State` column as the `SamplingKey` which would put all users from the same state in the same CrossValidation fold.